### ISSUE TITLE
Can this work for Godot 4+ games with multiplayer and open world chunks?

### ISSUE THREAD

**Issue by WithinAmnesia** [Bioblaze/godot_mmo_client#1]
Test chunks link: https://github.com/WithinAmnesia/ARPG/discussions/15

I'm trying to find a way to seamless load and unload chunks for a 2D multiplayer game project to make an open world with a working server using Godot 4.2.1.NET.

How can this work for multiplayer and what is needed for this to potentially work? What options can be used for chunk loading and unloading seamlessly in Godot 4.2.1.NET? Please give feedback.

**Comment by Bioblaze** [comment]
yes this could do that, but you would need to edit the code slightly on the server side

**Comment by WithinAmnesia** [comment]
Here is an update since I last posted.
https://jonathaneeckhout.itch.io/jdungeon For the goal right now it is easy to play this but imagine it has seamless chunk loading and unloading. As instead of the black world boundary a new chunk loads in seamlessly. What are your suggestions and thoughts?

Update: https://github.com/WithinAmnesia/ARPG/tree/ARPG-Infinite-Worlds Here is the 128x128 with 32x32 pixel tile chunk to test.

My initial testing seems to feel even faster combat with the 128x128 chunk. This 128x128 chunk has the same amount of entities as the 256x256 chunk. For I moved all of the entities over into the 128x128 chunk. This 128x128 chunk is a good test for using the Infinite-Worlds using the BinarySerializer for Godot 4.2+ from Theraot: https://gist.github.com/theraot/31515e28e2d8bfea33f6c6d5bcd852f6 . There needs to be some testing how to make the chunks seamlessly load and unload. https://gamedev.stackexchange.com/questions/209002/looking-for-help-godot-4-multiplayer-seamless-open-world-chunks

**Comment by WithinAmnesia** [comment]
> yes this could do that, but you would need to edit the code slightly on the server side

This is Amazing! What are your thoughts on the code by Theraot and the update with the 128x128 test chunk to download?

I am really hoping to find any help with this difficult puzzle and any insights and solutions and questions are super helpful for this is the one area I need aid with and if this can be helped it would help so many people and I feel its really important to do my best and listen to experts who are far better exerts herein than I am lol.

So any questions or initial thoughts? The demo can be downloaded and tinkered with if you are curious too? I made it 128x128 tiles with 32x32 pixel tiles to test so it hopefully 'tiles' makes chunk grids and keep the combat / server running fast better with seamless chunk loading and unloading.

I super want to have this happen but I hope I can help and do work and find good solutions and make friends, I just need a bit of guidance. I can put the time in and lift as much weight as possible. It's just what should be done and what areas / things should be pursued to help with the seamless chunk loading and unloading?

The code by Theraot looks really promising its just how can it best put to use and or what things also should be considered and work on? There's so many questions lol. I hope you know more than I do I am a bit puzzled right now lol. I really hope to help the game dev community with this big breakthrough.

What do you feel and what are your first thoughts and questions and ideas on what should be done next too? What should be done? All feedback is welcome.

**Comment by WithinAmnesia** [comment]
> yes this could do that, but you would need to edit the code slightly on the server side

The Github has Godot 3.XX how do we get this to work in Godot 4?
